### ISSUE TITLE
Adding semicolon after conditional block in a function causes error

### ISSUE THREAD

**Issue by mknadler** [sass/libsass#1536]
Kind of a weird problem. Bad syntax => an error in libsass but not ruby sass.
Tested on libsass 3.2.5, ruby sass 3.4.14.

```
@function test() {
  @if 1 < 2 { 
    @return 'test' 
  };
}
```

That semicolon shouldn't be there in the first place, and there should probably be one after `test`, but:
1. This doesn't cause an error in Ruby Sass
2. The error message is nonintuitive -- `only variable declarations and control directives are allowed inside functions on line X at column Y`

**Comment by mgreter** [comment]
Cannot reproduce on [latest master] [1]!

[1]: http://libsass.ocbnet.ch/srcmap/#QGZ1bmN0aW9uIHRlc3QoKSB7CiAgQGlmIDEgPCAyIHsgCiAgICBAcmV0dXJuICd0ZXN0JyAKICB9Owp9

**Comment by mknadler** [comment]
Weird & cool!

**Comment by mgreter** [comment]
Closing this as this will be fixed in upcoming 3.3 release!

**Comment by saper** [comment]
Yes, worked for me too. Calling this function also works properly.
